agent/src/main/scala/za/co/absa/atum/agent/dispatcher/Dispatcher.scala

```diff
@@ -18,8 +18,20 @@ package za.co.absa.atum.agent.dispatcher
 
 import za.co.absa.atum.model.dto.{AtumContextDTO, CheckpointDTO, PartitioningSubmitDTO}
 
+/**
+ *  This trait provides a contract for different dispatchers
+ */
 trait Dispatcher {
+  /**
+   *  This method is used to fetch AtumContext from server.
+   *  @param partitioning: PartitioningSubmitDTO to be used for fetching AtumContext.
```

Comment:
If above accepted, this should be reworded too, IMHO.

Reply:
Changed as per your suggestion.